tests/json_gen/generator.py

```diff
@@ -29,6 +31,57 @@ def _check_value_types(self, value_types) -> tuple:
         return value_types
 
 
+def update(d, u):
+    for k, v in u.items():
+        if isinstance(v, dict):
+            d[k] = update(d.get(k, {}), v)
+        elif isinstance(v, list):
+            d[k] = list(set(d.get(k, []) + v))
+        else:
+            if 'min' in k:
+                d[k] = max(v, d.get(k, v))
+            elif 'max' in k:
+                d[k] = min(v, d.get(k, v))
+            else:
+                d[k] = v
+    return d
+
+
+def difference(schema, chosen):
+    remove_keys = []
+    for key, schema_value in schema.items():
+        chosen_value = chosen.get(key)
+        if chosen_value is not None:
+            if isinstance(schema_value, dict):
+                schema[key] = difference(schema_value, chosen_value)
+            elif isinstance(schema_value, list):
+                schema[key] = [i for i in schema_value if i not in chosen_value]
+            else:
+                remove_keys.append(key)
+    for key in remove_keys:
+        schema.pop(key)
+    return schema
+
+
+def remove(d, u):
+    for k, v in u.items():
+        dv = d.get(k)
+        if dv is not None:
+            if k == 'required':
+                for req in v:
+                    d['properties'].pop(req, None)
+                d[k] = [i for i in d[k] if i not in v]
+            elif isinstance(v, dict):
+                d[k] = remove(dv, v)
+            elif isinstance(v, list):
+                d[k] = [i for i in dv if i not in v]
+    return d
+
+
+def sanitize(regex: str) -> str:
```

Comment:
I don't fully understand the purpose of this. You need to make a habit of explaining non-obvious things. This is only used once and seems to have a very specific purpose. Make that cohesion explicit by moving this function closer to the call site, potentially as a nested function.

A few more notes: `.` becomes a literal character inside brackets. What about dots that are already escaped? Last but not least take a look at re.escape(). All this under the preamble that I don't actually understand the issue this is supposed to address.